File: javascript/Valid Parentheses.js
function validParentheses(parenStr) {
  // Your code here
  
  const stack = [];
  const map = { ')' : '(' }
  
  for( let char of parenStr){
    
    
    if(char in map){
      
      if(stack.length === 0) return false
      
      let top = stack.pop()
    
      if(top !== map[char]){
        return false
      }
    }
    
    else{
      stack.push(char)
    }
    
  }
  return stack.length == 0
}

File: python/Valid Parentheses.py
def valid_parentheses(paren_str):
    
    stack = []
    dict = { ')' : '('}
    
    for char in paren_str:
        
        if char in dict:
            
            if len(stack) == 0:
                return False
            
            top = stack.pop()
            
            if top != dict[char]:
                return False
            
        else:
            stack.append(char)
        
    return len(stack) == 0